templates/rustdoc/header.html

```diff
@@ -191,6 +191,16 @@
                             </a>
                         </li>
 
+                    {# If the crate is a pre-release, display a warning and offer a link to the latest stable release#}
+                    {%- elif is_prerelease -%}
+                    <li class="pure-menu-item">
+                        <a href="{{ latest_path | safe }}" class="pure-menu-link warn"
+                           title="You are seeing a pre-release version of {{ krate.name }} crate. Click here to go to latest stable version.">
```

Comment:
I don't think this is quite right ... You could be on a pre-release version with no stable version available. I think this block should be inside `{%- elif not is_latest_version -%}` and only change the text, not the behavior.

Reply:
`{%- elif not is_latest_version -%}` is already the condition for the normal `Go to latest` message.  
I think 
```
{# If the crate is a pre-release, display a warning and offer a link to the latest stable
release. If a crate only has pre-release the warning won't be displayed for the latest #}
{%- elif is_prerelease and not is_latest_version -%}
```
is what we want. This won't display a warning if the user views the latest pre-release for a crate that has no stable releases but displays it for all other pre-releases.

Reply:
Yes, that sounds right. If you put it inside `{%- elif not is_latest_version -%}` you only have to change the text displayed without having to copy the HTML around it.

Reply:
I haven't used Tera a lot. I know how to use the condition for the text inside the span but I don't know how to do that elegantly for the title attribute. I'm also not sure if it would make the template cleaner if I somehow used interpolation for the title.

Reply:
Like this:
```html
                    {# If this isn't the most recent release, offer a link to the latest #}
                    {%- elif not is_latest_version -%}
                      {%- if not is_latest_version -%}
                        {% set title = "You are seeing an outdated version of the " ~ {{ krate.name }} ~ "crate. Click here to go to latest version."
                      {%- else %}
                        {% set title = "You are seeing a pre-release version of the " ~ {{ krate.name }} ~ "crate. Click here to go to latest stable version."
                      {%- endif %}
                        <li class="pure-menu-item">
                            <a href="{{ latest_path | safe }}" class="pure-menu-link warn"
                                title="{% title %}">
                                {{ "exclamation-triangle" | fas(fw=true) }}
                                <span class="title"> Go to latest version</span>
                            </a>
                        </li>
                    {%- endif -%}
```

Reply:
but if you think it's cleaner the other way, that's fine too, I have a tendency to DRY too much sometimes

Reply:
I've changed it and included the `not is_latest_version and krate.yanked` variant as well.